### PR TITLE
Allow string initializer for enums

### PR DESCRIPTION
**Pull request by zengjie**
String enum is a useful language feature, I think we should support it.

### CODE REVIEW

**Comment by Perryvw** [comment]
What about testcase:
```ts
enum abc {
    a,
    b = "def",
    c,
}

**Comment by zengjie** [comment]
Did you mean create a new test case with this input? or just modify the current one?

**Comment by Perryvw** [comment]
I meant adding a new testcase with that specific enum, it is not allowed in typescript and should error out (which you can test)

**Comment by Perryvw** [comment]
Also you probably shouldn't change old tests, just add some new ones.

**Comment by zengjie** [comment]
Yes. You are right, this test case will result an error.

**Comment by zengjie** [comment]
@Perryvw I've updated the test cases.

File: test/translation/lua/enum.lua [hunk 1]
```diff
@@ -1,4 +1,5 @@
 TestEnum={}
 TestEnum.val1=0
 TestEnum.val2=2
-TestEnum.val3=3
\ No newline at end of file
+TestEnum.val3=3
+TestEnum.val4="bye"
```
Comment (Perryvw) [inline test/translation/lua/enum.lua]: Revert this file back to just a generic enum as it was before, this case is covered by enumHeterogeneous.lua

File: test/translation/ts/enum.ts [hunk 2]
```diff
@@ -1,5 +1,6 @@
 enum TestEnum {
     val1 = 0,
     val2 = 2,
-    val3
-}
\ No newline at end of file
+    val3,
+    val4 = "bye"
+}
```
Comment (Perryvw) [inline test/translation/ts/enum.ts]: Also revert this file

**Comment by zengjie** [comment]
@Perryvw I've reverted the test case. 😄

**Pull request merged**